lib/ceased_member_row.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+require 'scraped'
+
+class CeasedMemberRow < Scraped::HTML
+  field :source do
+    noko.css('a @href').text
+  end
+
+  field :name do
+    noko.css('a').text
+  end
+
+  field :party do
+    # The layout is inconsistent. Scraping this way
+    # we always get the party name for each row
+    noko.xpath('text()')[0..1].text.split('.').last.tidy
+  end
+
+  field :end_date do
+    Date.parse(noko.text.match(/(?:\d\d?\.\d\d?.\d\d\d\d)/)[0]).to_s
```

Comment:
1. I can't work out why you're starting this with a non-capturing `?:`
2. You're quoting the '.' after the first group of digits, but not the second one: this should be consistent
2. A big chain of `\d`s like this, especially where some of them are actually `\d?` is quite hard to read. Being explicit with counts might be more legible: `\d{1,2}\.\d{1,2}\.\d{4}`
3. `text.match(/…/)[0]` is a little easier to follow (and is certainly more consistent with how we do it elsewhere), as `text[/…/, 1]`